src/Compilers/Core/Portable/InternalUtilities/InterlockedOperations.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```

Comment:
In the Workspaces layer we have "LazyInitialization", but it's not quite the same. I couldn't find helpers like this elsewhere and I needed it in both VB and C# so it made sense to put it in the core assembly. Actually, now that we have this, I think a lot of our thread-safe lazy-init code could benefit from these helpers - especially where we use Interlocked.CompareExchange and immediately read the field back.